### DISCUSSION TITLE
Is there any way to change log format?

### DISCUSSION THREAD

**Discussion by moooowoong** [acassen/keepalived#2406] [Q&A]
Hello members.

Is there any way to change keepalived's log format to json?

I'm using keepalived 2.2.8 on on-premise cluster.
Except keepalived, the other components uses json format to print log.

However, keepalived prints doesn't follow above policy. So I can not use json formatter :(

I looked around keepalived community but I couldn't find a solution. 
Please let me know if you know the way.

**Comment by pqarmitage** [comment]
keepaiived normally sends log messages in plain text format to syslog. There is a compile time (with associated run time) option to write the log messages to a file; this is really intended for debug purposes. There is no native, to keepalived, option to write the log messages in json format.

I suspect that it would be possible to have keepalived log to a file and have another process monitor that file (using inotify()) and write new log entries to another file in json format. That is not something that would be part of keepalived, since it would be a generic utility that could be used with any file.

Since all log messages are written though `log_message()` in lib/logger.c, in theory it should not be too hard to add an option to write the log messages in json format. If you can take a sample of keepalived log messages, and post them here in the current format and the same messages in the json format you would like, then we could have a look at adding an option for writing log messages in json format (I presume that would be to a file rather than to syslog, but let me know if that assumption is wrong).
